src/identity-providers/routes/EditMapper.ts

```diff
@@ -0,0 +1,26 @@
+import type { LocationDescriptorObject } from "history";
+import { generatePath } from "react-router-dom";
+import type { RouteDef } from "../../route-config";
+import { AddMapper } from "../add/AddMapper";
+
+export type IdentityProviderTab = "mappers";
+
+export type IdentityProviderEditMapperParams = {
+  realm: string;
+  id?: string;
+  tab?: IdentityProviderTab;
+  alias?: string;
+  providerId?: string;
```

Comment:
All these fields are required in the route path, so they should be required in this type as well. The `tab` field is never used in the path, so it can be removed. It is also a good idea to keep the order of these fields the same as they appear in the route path.

```suggestion
  providerId: string;
  alias: string;
  id: string;
```